### ISSUE TITLE
Can't perform a React state update on an unmounted component

### ISSUE THREAD

**Issue by Danetag** [hudovisk/react-optimize#35]
Hey team,

I've setup a basic`<Experiment>` with 2 `<Variant>` on a page:

```.jsx
const DefaultVariant = () => (
     <p>Default Variant</p>
)

const defaultV = <DefaultVariant />

<Experiment id="id-experiment" loader={defaultV}>
          <Variant id="0">
             { defaultV }
          </Variant>
          <Variant id="1">
            <p>Variant id1</p>
          </Variant>
</Experiment>
```

But when I quickly switch between pages, a React error shows up:

```
Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount method.
    in Experiment (at pages/index.js:39)
```

My guess is that it's due to the `updateVariantFromGlobalState()`,  but I might be wrong. Or is it something I'm missing?

Thanks!

**Comment by hudovisk** [comment]
Hi @Danetag! I guess you are correct, looking through the code I found that there was a timeout  not being cleared on `componentWillUnmount()`. I have released [v2.2.2](https://github.com/hudovisk/react-optimize/releases/tag/v2.2.2), please let me know if it fixes the problem.

**Comment by Danetag** [comment]
@hudovisk Thank you, it seems to resolve the issue!
